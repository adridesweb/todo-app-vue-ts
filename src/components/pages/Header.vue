<template>
  <header class="header">
    <ToggleCheckTasksVue v-if="!isTodoListEmpty" />
    <NewTask />
  </header>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { useTodosStore } from "@/composables";

import NewTask from "../NewTask.vue";
import ToggleCheckTasksVue from "../ToggleCheckTasks.vue";

export default defineComponent({
  name: "Header",
  components: { NewTask, ToggleCheckTasksVue },
  setup() {
    const { isTodoListEmpty } = useTodosStore();
    return {
      isTodoListEmpty,
    };
  },
});
</script>

<style scoped>
.header {
  position: relative;
  display: flex;
  align-items: center;
}
</style>
