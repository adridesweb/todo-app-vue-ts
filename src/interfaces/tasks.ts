export interface Task {
  id: string | number;
  title: string;
  completed: boolean;
  editing: boolean;
}
