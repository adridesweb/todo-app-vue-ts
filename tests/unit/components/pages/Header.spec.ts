import { mount } from "@vue/test-utils";
import { v4 as uuid } from "uuid";
import { Task } from "@/interfaces/tasks";

import Header from "@/components/pages/Header.vue";
import createVuexStore from "../../mock-data/mock-store";

describe("tests on pages/Header.vue", () => {
  const newBasicTask = <Task>{
    id: uuid(),
    title: "New task",
    completed: false,
    editing: false,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly with one task in store", () => {
    const store = createVuexStore({ tasks: [newBasicTask] });
    const wrapper = mount(Header, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
