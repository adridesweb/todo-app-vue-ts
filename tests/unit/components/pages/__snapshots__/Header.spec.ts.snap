// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on pages/Header.vue renders correctly with one task in store 1`] = `
<header
  class="header"
>
  
  <input
    class="toggle-all"
    type="checkbox"
  />
  <label
    for="toggle-all"
  >
     Mark all as complete 
  </label>
  
  <input
    autofocus=""
    class="new-todo"
    placeholder="What needs to be done?"
  />
</header>
`;
